SimDataFormats/CaloAnalysis/interface/SimTauCPLink.h

```diff
@@ -0,0 +1,82 @@
+#ifndef SimDataFormats_SimTauCPLink_h
+#define SimDataFormats_SimTauCPLink_h
+
+#include "DataFormats/HGCalReco/interface/TICLCandidate.h"
+#include "SimDataFormats/CaloAnalysis/interface/CaloParticleFwd.h"
+#include "SimDataFormats/CaloAnalysis/interface/CaloParticle.h"
+#include "SimDataFormats/CaloAnalysis/interface/SimCluster.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+class SimTauCPLink {
+public:
+  SimTauCPLink(){};
+  ~SimTauCPLink(){};
+  struct DecayNav {
+    int pdgId_;
+    int resonance_idx_;
+    int calo_particle_idx_;
+    int gen_particle_idx_;
+    int pdgId() const { return pdgId_; }
+    int resonance_idx() const { return resonance_idx_; }
+    int calo_particle_idx() const { return calo_particle_idx_; }
+    int gen_particle_idx() const { return gen_particle_idx_; }
+  };
+
+  std::vector<std::pair<int, int>> resonances;
+  std::vector<DecayNav> leaves;
+  CaloParticleRefVector calo_particle_leaves;
+  int decayMode;
+
+  enum decayModes {
```

Comment:
You defined the very same enum in another place. If you bring the `buildDecayModes` method part of this class, you can avoid this duplication.

Reply:
@agrubercms please comment.

Reply:
Moved `buildDecayModes` from `SimTauProducer` to the `SimTauCPLink` class in last commit to avoid this duplication.